tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,314 @@
+    figure at once, and returns an object array with handles for
+    the axes in the grid.  See also `.Figure.subplots`.
+
+or
+
+`~matplotlib.pyplot.subplot_mosaic`
+    A more flexible function used to create figures and a grid of axes,
+    with the added flexibility that some of the axes can span rows or
+    columns, and that the axes are returned in a labelled dictionary instead
+    of an array.  See also `.Figure.subplot_mosaic`.
+
+Sometimes it is natural to have more than one distinct group of axes grids,
+in which case Matplotlib has the concept of `~.figure.SubFigure`:
+
+`~matplotlib.figure.SubFigure`
+    A virtual figure within a figure.
+
+Underlying these is the concept of a `~.gridspec.GridSpec` and
+`~.SubplotSpec`:
+
+`~matplotlib.gridspec.GridSpec`
+    Specifies the geometry of the grid that a subplot will be
+    placed. The number of rows and number of columns of the grid
+    need to be set. Optionally, the subplot layout parameters
+    (e.g., left, right, etc.) can be tuned.
+
+`~matplotlib.gridspec.SubplotSpec`
+    Specifies the location of the subplot in the given `.GridSpec`.
+
+An older way to add axes in grids:
+
+`~matplotlib.pyplot.subplot2grid`
+    A helper function that is similar to `.pyplot.subplot`,
+    but uses 0-based indexing and let subplot to occupy multiple cells.
+    This function is not covered in this tutorial.
+
+.. redirect-from:: /tutorials/intermediate/gridspec
+
+"""
+
+import matplotlib.pyplot as plt
+
+import numpy as np
+
+############################################################################
+# High-level methods for making grids
+# ===================================
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can create a basic 2-by-2 grid of axes using
+# :func:`~matplotlib.pyplot.subplots`.  It returns a
+# :class:`~matplotlib.figure.Figure` instance and an array of
+# :class:`~matplotlib.axes.Axes` objects.  The axes objects can
+# be used to access methods to place artists on the axes; here we
+# use `~.Axes.annotate`, but other examples could be `~.Axes.plot`,
+# `~.Axes.pcolormesh`, etc.
+
+fig, axs = plt.subplots(ncols=2, nrows=2, figsize=(4.5, 3.5),
+                        constrained_layout=True)
+for row in range(2):
+    for col in range(2):
+        axs[row, col].annotate(f'axs[{row}, {col}]', (0.1, 0.5),
+                               xycoords='axes fraction', va='center')
+fig.suptitle('subplots')
+
+##############################################################################
+# The same effect can be achieved with `~.pyplot.subplot_mosaic`,
+# but the return type is a dictionary instead of an array, where the user
+# can give the keys useful meanings.  Here we provide two lists, each list
+# representing a row, and each element in the list a key representing the
+# column.  Note that keys can be any dictionary key, but we typically use
+# strings:
+
+fig, axs = plt.subplot_mosaic([['upleft', 'upright'], ['loleft', 'loright']],
```

Comment:
```suggestion
fig, axd = plt.subplot_mosaic([['upleft', 'upright'],
                               ['loleft', 'loright']],
```
- Make the 2D list look similar to the layout.
- `axd` the canonical name according to the [tutorial](https://matplotlib.org/devdocs/tutorials/provisional/mosaic.html.)

(Also in all following examples)